src/main/java/org/jabref/gui/bibtexkeypattern/BibtexKeyPatternPanel.java

```diff
@@ -153,14 +183,61 @@ private void buildGUI() {
         gbl.setConstraints(btnDefaultAll, con);
         btnDefaultAll.addActionListener(e -> {
             // reset all fields
-            for (JTextField field : textFields.values()) {
-                field.setText("");
+            for (TextField textField : textFieldArray) {
+                textField.setText("");
             }
 
             // also reset the default pattern
             defaultPat.setText((String) Globals.prefs.defaults.get(JabRefPreferences.DEFAULT_BIBTEX_KEY_PATTERN));
         });
         add(btnDefaultAll);
+
+        Button btnDefaultAll1 = new Button(Localization.lang("Reset all"));
+        btnDefaultAll1.setFont(font1);
+        btnDefaultAll1.setOnAction(e-> {
+            // reset all fields
+            for (JTextField field : textFields.values()) {
+                field.setText("");
+            }
+            textField.setText((String) Globals.prefs.defaults.get(JabRefPreferences.DEFAULT_BIBTEX_KEY_PATTERN));
+        });
+        gridPane.add(btnDefaultAll1,3,24);
+    }
+
+    private void addExtraText() {
+        Label []label = new Label[19];
+        Button []button = new Button[19];
+        for (int i = 0; i <= 18; i++) {
+            textFieldArray[i] = new TextField();
+            button[i] = new Button("Default");
+            button[i].setFont(new javafx.scene.text.Font(10));
+            button[i].setOnAction(e-> defaultPat.setText((String) Globals.prefs.defaults.get(JabRefPreferences.DEFAULT_BIBTEX_KEY_PATTERN)));
+        }
+        label[0] = new Label("Article");
```

Comment:
Instead of hard-coding these fields, you can use a foreach loop like `for (EntryType type : EntryTypes.getAllValues(mode)) `